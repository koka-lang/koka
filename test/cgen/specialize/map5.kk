
public fun test(xs : list<a>, y : int, f : a -> int) {
  xs.map(fn(x){ f(x) + y })   // free variable  
}

public fun main() {
  list(1,10).test(5, inc).sum.println
}
