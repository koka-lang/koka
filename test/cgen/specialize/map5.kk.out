115
 
cgen/specialize/map5/.lift000-test: forall<a> (f : (a) -> int, y : int, xs0 : list<a>) -> list<int>
cgen/specialize/map5/.lift000-main: (xs : list<int>) -> list<int>
cgen/specialize/map5/main: () -> console ()
cgen/specialize/map5/test: forall<a> (xs : list<a>, y : int, f : (a) -> int) -> list<int>